bloptools/test_functions.py

```diff
@@ -13,11 +13,43 @@ def himmelblau(x1, x2):
     return (x1**2 + x2 - 11) ** 2 + (x1 + x2**2 - 7) ** 2
 
 
+def constrained_himmelblau(x1, x2):
+    if x1**2 + x2**2 > 50:
+        return np.nan
+    return himmelblau(x1, x2)
+
+
+def skewed_himmelblau(x1, x2):
+    _x1 = 2 * x1 + x2
+    _x2 = 0.5 * (x1 - 2 * x2)
+
+    return constrained_himmelblau(_x1, _x2)
+
+
+def bukin(x1, x2):
+    return 100 * np.sqrt(np.abs(x2 - 1e-2 * x1**2)) + 0.01 * np.abs(x1)
```

Comment:
All these functions need a docstring with a reference to Wikipedia or other resources.